### PR TITLE
4.1.1 updates

### PR DESCRIPTION
**Pull request by krzywon**
The changes made for release 4.1.1 and 4.1.2 were made in separate branches and have not been integrated into trunk as of yet. These two point releases fixed [918](http://trac.sasview.org/ticket/918), [929](http://trac.sasview.org/ticket/929), [938](http://trac.sasview.org/ticket/938), [948](http://trac.sasview.org/ticket/948), [960](http://trac.sasview.org/ticket/960), and [984](http://trac.sasview.org/ticket/984). Most of these tickets have already been closed.

I have also added a fix for tickets [978](http://trac.sasview.org/ticket/978) and [985](http://trac.sasview.org/ticket/985) into this branch.

* Closes ticket [918](http://trac.sasview.org/ticket/918)
* Closes ticket [929](http://trac.sasview.org/ticket/929)
* Closes ticket [938](http://trac.sasview.org/ticket/938)
* Closes ticket [948](http://trac.sasview.org/ticket/948)
* Closes ticket [960](http://trac.sasview.org/ticket/960)
* Closes ticket [984](http://trac.sasview.org/ticket/984)
* Closes ticket [978](http://trac.sasview.org/ticket/978)
* Closes ticket [985](http://trac.sasview.org/ticket/985)

### CODE REVIEW

File: src/sas/sasgui/perspectives/fitting/fitpanel.py [hunk 1]
```diff
@@ -394,17 +400,14 @@ def delete_data(self, data):
...
-                    self.on_close_page(event=None)
-                    temp = self.GetSelection()
-                    self.DeletePage(temp)
+                    self.close_page_with_data(temp_data)
             if self.sim_page is not None:
                 if len(self.sim_page.model_list) == 0:
                     pos = self.GetPageIndex(self.sim_page)
                     self.SetSelection(pos)
                     self.on_close_page(event=None)
                     temp = self.GetSelection()
```
Comment (pkienzle) [inline src/sas/sasgui/perspectives/fitting/fitpanel.py]: temp no longer used?

File: src/sas/sascalc/dataloader/readers/cansas_reader_HDF5.py [hunk 2]
```diff
@@ -135,6 +136,7 @@ def read_children(self, data, parent_list):
                     self._initialize_new_data_set(parent_list)
                 # Recursion step to access data within the group
                 self.read_children(value, parent_list)
+                self.parent_class = parent_class
```
Comment (pkienzle) [inline src/sas/sascalc/dataloader/readers/cansas_reader_HDF5.py]: self.parent_class assigned twice?

**Comment by butlerpd** [comment]
Ticket #978 seems to be fixed but the errors reported in ticket #985 still seem to be a problem.  Also ticket #984 seems to be throwing a longer stream of information though that could be due to the development environment.  This may need an installer to properly test?

**Comment by rozyczko** [comment]
Ready for testing on Linux

**Comment by rozyczko** [comment]
Ready for testing on Win32
Ready for testing on OSX

**Comment by rozyczko** [comment]
Ready for testing on Win32

**Comment by rozyczko** [comment]
ready for testing on osx

**Comment by butlerpd** [comment]
So using the installer version of this branch it looks like somebody left a bunch of debugging code from trying to find why the PDF was not saving? the output in the logfile every time a save report to PDF is run is:
```
2017-09-19 20:47:42,809 : DEBUG : xhtml2pdf (document.pyc:81) :: pisaDocument options:
  src = u'<html>\n<head>\n<meta http-equiv=Content-Type content=\'text/html; charset=windows-1252\'> \n<meta name=Generator >\n</head>\n<body lang=EN-US>\n<div class=WordSection1>\n<p class=MsoNormal><b><span ><center><font size=\'4\' >           latex particles 0.5micron diameter in D2O Qdev [ Sep 19 2017 20;38;41 ]</font></center></span></center></b></p><p class=MsoNormal>&nbsp;</p>\n<p class=MsoNormal><center><font size=\'4\' > File name:            latex_smeared.xml \n</font></center></p>\n<p class=MsoNormal><center><font size=\'4\' > Model name: sphere \n</font></center></p>\n<p class=MsoNormal><center><font size=\'4\' > Q Range:     min =  0.003797, max =  0.401 \n</font></center></p>\n<p class=MsoNormal><center><font size=\'4\' > Chi2/Npts =  46.049 \n</font></center></p>\n<p class=MsoNormal>&nbsp;</p>\n<p class=MsoNormal><center><font size=\'4\' >  scale  =  0.0069012 +- 9.9301e-06  \n</font></center></p>\n<p class=MsoNormal><center><font size=\'4\' >  background  =  0.060165 +- 4.1352e-05   1/cm \n</font></center></p>\n<p class=MsoNormal><center><font size=\'4\' >  sld  =  1  1e-6/Ang^2 \n</font></center></p>\n<p class=MsoNormal><center><font size=\'4\' >  sld_solvent  =  6  1e-6/Ang^2 \n</font></center></p>\n<p class=MsoNormal><center><font size=\'4\' >  radius  =  1971.7 +- 1.6587   Ang \n</font></center></p>\n\n<p class=MsoNormal>&nbsp;</p>\n\n<p class=MsoNormal>&nbsp;</p>\n<br>\n<p class=MsoNormal><b><span ><center> <font size=\'4\' > Graph\n</font></span></center></b></p>\n<p class=MsoNormal>&nbsp;</p>\n<center>\n<br><font size=\'4\' >Model Computation</font>\n<br><font size=\'4\' >Data: "           latex particles 0.5micron diameter in D2O Qdev [ Sep 19 2017 20;38;41 ]"</font><br>\n\n\n<img src="C:\\Users\\paperspace\\Documents\\SasViewOutputTests\\test_img0.png" >\n</img>\n<p class=MsoNormal>&nbsp;</p>\n<img src="C:\\Users\\paperspace\\Documents\\SasViewOutputTests\\test_img1.png" >\n</img>\n<p class=MsoNormal>&nbsp;</p>\n</center>\n</div>\n</body>\n</html>\n'
  dest = <open file u'C:\\Users\\paperspace\\Documents\\SasViewOutputTests\\test.pdf', mode 'w+b' at 0x2101BD88>
  path = None
  link_callback = None
  xhtml = False
2017-09-19 20:47:42,884 : DEBUG : xhtml2pdf (util.pyc:517) :: FileObject 'C:\\Users\\paperspace\\Documents\\SasViewOutputTests\\test_img0.png', Basepath: 'C:\\Users\\paperspace\\Documents\\SasViewOutputTests\\__dummy__'
2017-09-19 20:47:42,884 : DEBUG : xhtml2pdf (util.pyc:532) :: URLParts: ParseResult(scheme='c', netloc='', path='\\Users\\paperspace\\Documents\\SasViewOutputTests\\test_img0.png', params='', query='', fragment='')
2017-09-19 20:47:43,119 : DEBUG : PIL.PngImagePlugin (PngImagePlugin.pyc:135) :: STREAM IHDR 16 13
2017-09-19 20:47:43,119 : DEBUG : PIL.PngImagePlugin (PngImagePlugin.pyc:135) :: STREAM sBIT 41 4
2017-09-19 20:47:43,119 : DEBUG : PIL.PngImagePlugin (PngImagePlugin.pyc:511) :: sBIT 41 4 (unknown)
2017-09-19 20:47:43,119 : DEBUG : PIL.PngImagePlugin (PngImagePlugin.pyc:135) :: STREAM pHYs 57 9
2017-09-19 20:47:43,119 : DEBUG : PIL.PngImagePlugin (PngImagePlugin.pyc:135) :: STREAM IDAT 78 8192
2017-09-19 20:47:43,122 : DEBUG : xhtml2pdf (util.pyc:517) :: FileObject 'C:\\Users\\paperspace\\Documents\\SasViewOutputTests\\test_img1.png', Basepath: 'C:\\Users\\paperspace\\Documents\\SasViewOutputTests\\__dummy__'
2017-09-19 20:47:43,122 : DEBUG : xhtml2pdf (util.pyc:532) :: URLParts: ParseResult(scheme='c', netloc='', path='\\Users\\paperspace\\Documents\\SasViewOutputTests\\test_img1.png', params='', query='', fragment='')
2017-09-19 20:47:43,124 : DEBUG : PIL.PngImagePlugin (PngImagePlugin.pyc:135) :: STREAM IHDR 16 13
2017-09-19 20:47:43,124 : DEBUG : PIL.PngImagePlugin (PngImagePlugin.pyc:135) :: STREAM sBIT 41 4
2017-09-19 20:47:43,124 : DEBUG : PIL.PngImagePlugin (PngImagePlugin.pyc:511) :: sBIT 41 4 (unknown)
2017-09-19 20:47:43,124 : DEBUG : PIL.PngImagePlugin (PngImagePlugin.pyc:135) :: STREAM pHYs 57 9
2017-09-19 20:47:43,124 : DEBUG : PIL.PngImagePlugin (PngImagePlugin.pyc:135) :: STREAM IDAT 78 8192
2017-09-19 20:47:43,134 : DEBUG : PIL.PngImagePlugin (PngImagePlugin.pyc:135) :: STREAM IHDR 16 13
2017-09-19 20:47:43,134 : DEBUG : PIL.PngImagePlugin (PngImagePlugin.pyc:135) :: STREAM sBIT 41 4
2017-09-19 20:47:43,134 : DEBUG : PIL.PngImagePlugin (PngImagePlugin.pyc:511) :: sBIT 41 4 (unknown)
2017-09-19 20:47:43,135 : DEBUG : PIL.PngImagePlugin (PngImagePlugin.pyc:135) :: STREAM pHYs 57 9
2017-09-19 20:47:43,135 : DEBUG : PIL.PngImagePlugin (PngImagePlugin.pyc:135) :: STREAM IDAT 78 8192
2017-09-19 20:47:43,181 : DEBUG : PIL.PngImagePlugin (PngImagePlugin.pyc:135) :: STREAM IHDR 16 13
2017-09-19 20:47:43,183 : DEBUG : PIL.PngImagePlugin (PngImagePlugin.pyc:135) :: STREAM sBIT 41 4
2017-09-19 20:47:43,183 : DEBUG : PIL.PngImagePlugin (PngImagePlugin.pyc:511) :: sBIT 41 4 (unknown)
2017-09-19 20:47:43,183 : DEBUG : PIL.PngImagePlugin (PngImagePlugin.pyc:135) :: STREAM pHYs 57 9
2017-09-19 20:47:43,183 : DEBUG : PIL.PngImagePlugin (PngImagePlugin.pyc:135) :: STREAM IDAT 78 8192
```

can we get rid of that before merging please?

**Comment by butlerpd** [comment]
Testing with the installer on windows 10 machine both ticket 978 AND 985 seem to now be fixed.  There may still be some loader issues to resolve but I think these two tickets should be marked for closure (along with ticket 984 ... and the rest of the tickets of course) when the request is merged (hopefully after removing the debug code?)

**Comment by butlerpd** [comment]
As part of loader issues I note that while it seems to load fine, if a project file includes a Pr analysis component the following errors get logged to the logfile (again no obvious errors in loading from user point of view)
```
2017-09-19 21:00:18,604 : ERROR : sas.sasgui.perspectives.pr.inversion_state (inversion_state.pyc:313) :: InversionState.fromXML: inconsistant number of coefficients: 0 10
2017-09-19 21:00:18,604 : ERROR : sas.sasgui.perspectives.pr.inversion_state (inversion_state.pyc:350) :: InversionState.fromXML: inconsistant dimensions of the  covariance matrix: 0 10
```

**Comment by pkienzle** [comment]
The PIL debug statements are coming from the PIL package. You will need to play with the settings in c:\Program Files (x86)\SasView\logging.ini to get it to do what you want.  I'm guessing that setting [logger_root] to level=WARNING will be sufficient.  Once you've got something you are happy with update the sasview/logging.ini file with the new settings.

**Comment by butlerpd** [comment]
Ok @pkienzle  I see that these PIL debug messages also appear when running from branch master and are thus NOT an issue introduced by this pull request.  Given this I agree that asking for this PR to fix that is out of scope (particularly since the point is only to pull in the changes from the last two point releases).  I will approve then merge this pull request.

**Pull request merged**